@here/harp-mapview/lib/ClipPlanesEvaluator.ts

```diff
@@ -0,0 +1,171 @@
+    private static readonly EYE_INVERSE = new THREE.Vector3(0, 0, -1);
+
+    evaluateClipPlanes(camera: Camera, projection: Projection): { near: number; far: number } {
+        const clipPlanes = super.evaluateClipPlanes(camera, projection);
+
+        // Tilt angle will be taken into consideration only for non-spherical projections,
+        // otherwise take super class values.
+        if (projection.type !== ProjectionType.Spherical) {
+            camera.getWorldDirection(this.m_tempVector3);
+            // tslint:disable-next-line:no-unused-variable
+            const angle = THREE.Math.radToDeg(
+                this.m_tempVector3.angleTo(TiltBasedClipPlanesEvaluator.EYE_INVERSE)
+            );
+            // TODO: Implement camera angle based planes interpolation.
+            // ...
+            // Old implementation was just clamping default (base) results,
+            // this gonna be replaced by something smarter:
+            clipPlanes.near = Math.max(clipPlanes.near, 20);
+            clipPlanes.far = Math.max(clipPlanes.far, 2000);
+
+            // Preserve limits inherited from default implementation.
+            clipPlanes.near = Math.max(this.nearMin, clipPlanes.near);
+            clipPlanes.far = Math.max(
+                clipPlanes.near * this.nearFarMultiplier + this.farOffset,
+                clipPlanes.far
+            );
+        }
+        return clipPlanes;
+    }
+}
+
+/**
+ * Provides the most basic evaluation concept giving fixed values with some constraints.
+ */
+export class FixedClipPlanesEvaluator implements ClipPlanesEvaluator {
+    readonly minFar: number;
+    private m_nearPlane: number;
+    private m_farPlane: number;
+
+    constructor(readonly minNear: number = 1, readonly minFarOffset: number = 10) {
+        this.minFar = minNear + minFarOffset;
+        this.m_nearPlane = minNear;
+        this.m_farPlane = this.minFar;
+    }
+
+    get nearPlane(): number {
+        return this.m_nearPlane;
+    }
+
+    set nearPlane(fixedNear: number) {
+        this.invalidatePlanes(fixedNear, this.m_farPlane);
+    }
+
+    get farPlane(): number {
+        return this.m_farPlane;
+    }
+
+    set farPlane(fixedFar: number) {
+        this.invalidatePlanes(this.m_nearPlane, fixedFar);
+    }
+
+    evaluateClipPlanes(camera: Camera, projection: Projection): { near: number; far: number } {
+        // We do not need to perform actual evaluation cause results are precomputed and
+        // kept stable until somebody changes the properties.
+        return { near: this.m_nearPlane, far: this.m_farPlane };
+    }
+
+    private invalidatePlanes(near: number, far: number) {
+        // When clamping prefer to extend far plane at about minimum distance, giving
+        // near distance setup priority over far.
+        const nearDist: number = Math.max(this.minNear, near);
+        const farDist: number = Math.max(Math.max(this.minFar, far), nearDist + this.minFarOffset);
```

Comment:
Math.max accepts any amount of arguments
```suggestion
        const farDist: number = Math.max(this.minFar, far, nearDist + this.minFarOffset);
```

Reply:
Ohh... good point! Will fix it.